apisix/core/config_etcd.lua

```diff
@@ -166,6 +177,39 @@ local function sync_data(self)
                           ", it shoud be a object")
             end
 
+            -- check plugins schema
+            local plugins_conf = item.plugins
+            if plugins_conf then
+                for name, plugin_conf in pairs(plugins_conf) do
```

Comment:
check the plugin `conf` in this way, it is not a good way.

config_etcd.lua is a basic component, it should not know the plugin.

I think we can export a new option when creating new `config_etcd` object, then we can add more check logic in `filter_func` .

Example:

```lua
core.config.new("/routes", {
    automatic = true,
    item_schema = core.schema.route,
    filter_func = function ()
        -- your code
    end
    filter = filter,
})
```